src/core/server/metrics/ops_metrics_event_schema.ts

```diff
@@ -0,0 +1,350 @@
+        _meta: {
+          description: 'heap memory usage',
+          optional: false,
+        },
+      },
+      event_loop_delay: {
+        type: 'long',
+        _meta: { description: 'mean event loop delay since last collection' },
+      },
+      event_loop_delay_histogram: {
+        properties: {
+          fromTimestamp: {
+            type: 'date',
+            _meta: {
+              description:
+                'The first timestamp the interval timer kicked in for collecting data points.',
+            },
+          },
+          lastUpdatedAt: {
+            type: 'date',
+            _meta: {
+              description:
+                'Last timestamp the interval timer kicked in for collecting data points.',
+            },
+          },
+          min: {
+            type: 'long',
+            _meta: { description: 'The minimum recorded event loop delay.' },
+          },
+          max: {
+            type: 'long',
+            _meta: { description: 'The maximum recorded event loop delay.' },
+          },
+          mean: {
+            type: 'long',
+            _meta: { description: 'The mean of the recorded event loop delays.' },
+          },
+          exceeds: {
+            type: 'long',
+            _meta: {
+              description:
+                'The number of times the event loop delay exceeded the maximum 1 hour event loop delay threshold.',
+            },
+          },
+          stddev: {
+            type: 'long',
+            _meta: { description: 'The standard deviation of the recorded event loop delays.' },
+          },
+          percentiles: {
+            properties: {
+              '50': {
+                type: 'long',
+                _meta: { description: '50th percentile of delays of the collected data points.' },
+              },
+              '75': {
+                type: 'long',
+                _meta: { description: '75th percentile of delays of the collected data points.' },
+              },
+              '95': {
+                type: 'long',
+                _meta: { description: '95th percentile of delays of the collected data points.' },
+              },
+              '99': {
+                type: 'long',
+                _meta: { description: '99th percentile of delays of the collected data points.' },
+              },
+            },
+            _meta: {
+              description: 'An object detailing the accumulated percentile distribution.',
+              optional: false,
+            },
+          },
+        },
+        _meta: {
+          description: 'node event loop delay histogram since last collection',
+          optional: false,
+        },
+      },
+      uptime_in_millis: {
+        type: 'long',
+        _meta: { description: 'uptime of the kibana process' },
+      },
+    },
+    _meta: {
+      description: 'Process related metrics.',
+      optional: false,
+    },
+  },
+  processes: {
+    type: 'array',
+    items: {
+      properties: {
```

Comment:
nit: this is the same as `process`. I wonder if we could reuse the schema. Also, do we really need to send the array here? Right now Kibana only supports one node... and you know how cool ES is with arrays of objects... 😉

Reply:
I have mixed feelings about this. Something tells me that the event shouldn't duplicate information:

On one hand, if `process` is deprecated, then maybe a future-proof solution is to not send it so folks don't create visualizations based on a field that will eventually disappear. In this case, I'd say we only send `processes`.

On the other hand, sending the `processes` array seems very unusable from the analytics point of view. Right now there's only one element in the array (and probably it will always be, based on the current state of the multiple initiatives for _clustering_ Kibana). But if we are looking at future-proving this solution, we'll need to account that there might be multiple elements in this array, so we'll need to unwrap it on the indexing side.

AFAIK, it's still unclear how we're going to do the _clustering_ (via Node.js `cluster` workers, separate processes with different configs in Docker containers or something else?), meaning we don't know whether each process will report its own telemetry (EBT in this case) or it will be centralized through a _main_ process. If the latter, are we confident that the shape of `processes` won't change?

IMO, those _unknowns_ makes it hard to identify what a future-proof solution would be. If I had to choose, I guess I'd go for NOT sending the deprecated field (it's harder to roll back to this field even if the final _clustering_ implementation means the array will always contain one element), and unwrapping the array from day one (potentially mapping it to `process`). 

What do you think?

Reply:
It's because of all the unknowns that I didn't want to change the data shape and decided to go for shipping the data as we handle it in the metrics service itself.
Monitoring uses the data as-is and I was thinking about being able to cross-correlate EBT data with metrics from monitoring and, potentially, the Kibana logs for ops metrics. 

That being said, we could use the schema description to make it clear we're unwrapping the single entry from `processes` here.

Reply:
@afharo I went with [mapping the single entry](https://github.com/elastic/kibana/pull/132298/files#diff-013ec112565649e8eabdfd4f26a2c36db9fc9751aa2f274487730717cbee91c5R100) in `processes` (if we have it) or the deprecated `process` field as a fallback to a new field to get past the deprecation warning.

Reply:
Thank you @TinaHeiligers! Sorry! I think my initial baffling got you confused... Apologies for that 🙏 

I really liked your intention of future-proofing this solution. For that reason, I think we only need to drop `process` and send `processes` as-is.

Then, on the indexing side, we can tell the indexer to unwrap the array (potentially mapping to `process` in singular). If Kibana ever sends more than one element in `processes`, it will split the event into as many documents as elements in this array. I think this will provide a full e2e-future-proof solution.

What do you think?

Reply:
Thanks for explaining! Dropping the deprecated field and sending the rest as-is to me is the best approach since we're not making any assumptions on what ops metrics Kibana collects.

Upstream work for handling the array format should be fine too, we'll see when we get there.